pkg/plugins/manager/signature/manifestverifier/verifier.go

```diff
@@ -111,20 +174,73 @@ func (pmv *ManifestVerifier) GetPublicKey(keyID string) (string, error) {
 	}()
 
 	if err := json.NewDecoder(resp.Body).Decode(&data); err != nil {
-		return "", err
+		return err
 	}
 
 	if len(data.Items) == 0 {
-		return "", errors.New("missing public key")
+		return errors.New("missing public key")
 	}
 
+	cachedKeys, err := pmv.kv.Keys(ctx, "")
```

Comment:
this is going to include the last_updated item, meaning that it (and any other extra k/v pairs in the namespace) will get deleted in the block below, we should prefix the k/v key for the keys so we can target them accurately.
```suggestion
	cachedKeys, err := pmv.kv.Keys(ctx, "key-")
```

Reply:
right, I missed this, thanks!